### DISCUSSION TITLE
Custom Criteria

### DISCUSSION THREAD

**Discussion by Leventeand2** [FabricMC/fabric-api#4098] [Mod Dev Support] [answered]
### Description

**Hey guys!** I'm trying to make an advancement where if a player walks on a block, the criteria is triggered, but I'm unable to determine how to. Can someone please help me? Thank you! _(I'm new to both Java and Fabric)_

**Comment by haykam821** [comment, marked as answer]
You can use the `minecraft:location` criterion to implement this behavior. The 'Light as a Rabbit' advancement in vanilla Minecraft has a similar requirement to what you describe (walking on powder snow with leather boots) which you can use as an example.

The following is the relevant part of the advancement JSON required for this behavior:

```json
{
	"criteria": {
		"walk_on_<your block name>": {
			"trigger": "minecraft:location",
			"conditions": {
				"player": [
					{
						"condition": "minecraft:entity_properties",
						"entity": "this",
						"predicate": {
							"stepping_on": {
								"block": {
									"blocks": "<your block id here>"
								}
							}
						}
					}
				]
			}
		}
	}
}
```

**Reply by Leventeand2** [reply]
Ohhhh Thanks!
